src/partition/ReceivedPartition.cpp

```diff
@@ -119,21 +119,25 @@ void ReceivedPartition::compute()
   PRECICE_DEBUG("Tag vertices for filtering: 1st round.");
   _mesh->computeState(); // normals need to be ready for NP mapping
   // go to both meshes, vertex is tagged if already one mesh tags him
-  if (_fromMapping)
-    _fromMapping->tagMeshFirstRound();
-  if (_toMapping)
-    _toMapping->tagMeshFirstRound();
+  for (mapping::PtrMapping fromMapping : _fromMappings) {
+    fromMapping->tagMeshFirstRound();
+  }
+  for (mapping::PtrMapping toMapping : _toMappings) {
+    toMapping->tagMeshFirstRound();
+  }
 
   // (3) Define which vertices are owned by this rank
   PRECICE_DEBUG("Create owner information.");
   createOwnerInformation();
 
   // (4) Tag vertices 2nd round (what should be filtered out)
   PRECICE_DEBUG("Tag vertices for filtering: 2nd round.");
-  if (_fromMapping)
-    _fromMapping->tagMeshSecondRound();
-  if (_toMapping)
-    _toMapping->tagMeshSecondRound();
+  for (mapping::PtrMapping fromMapping : _fromMappings) {
+    fromMapping->tagMeshSecondRound();
+  }
+  for (mapping::PtrMapping toMapping : _toMappings) {
+    toMapping->tagMeshSecondRound();
+  }
```

Comment:
We could refactor this into a separate function.
This would make this long function a bit more readable.

```suggestion
  tagMeshSecondRounds();
```